server/src/main/java/io/deephaven/server/table/TableServiceGrpcImpl.java

```diff
@@ -276,11 +248,115 @@ public void updateBy(UpdateByRequest request, StreamObserver<ExportedTableCreati
         oneShotOperationWrapper(BatchTableRequest.Operation.OpCase.UPDATE_BY, request, responseObserver);
     }
 
+    private Object getSeekValue(Literal literal, Class<?> dataType) {
+        if (literal.hasStringValue()) {
+            return literal.getStringValue();
```

Comment:
It actually works fine with the String value. See in SeekRow.java:243, it does a String comparison if a String is passed in. I don't think it should assert here - if you pass in a value as a String it can still seek to it. (Remember, I didn't build this API - I'm just porting it.) With an assertion here, this case wouldn't work:

```
await table.seekRow(0, table.columns[1], 'String', '9')
```

Ditto with nanotime - you can pass in a date as a String and it will compare against the data type.
```
var strDate = dh.DateWrapper.ofJsDate(new Date()).valueOf()
await table.seekRow(0, table.columns[0], 'Datetime', strDate)
```

I can just translate the exception here and throw a `GrpcUtil.statusRuntimeException`?

Reply:
I think this is the case we talked about over a call before, where yes, technically, it would compare the values, but it would compare them as strings, not numbers - so 100 < 99. For date values, that will only matter when our ns precision increases the decimal-encoded length, however this is more likely to happen for 'typical' numeric types.

Agreed that the ported "carq" is very permissive, I'm just saying that we should try to avoid that unless we have no other choice, and mark "this is probably wrong, and easy to be made right" cases more clearly for api consumers.

Reply:
I'm not sure when the case you're describing would get hit. I've filed an issue for numeric columns as it doesn't seem to go to the "closest" row for them: https://deephaven.atlassian.net/browse/DH-13839

I still want to just get this in to match what's in Enterprise, so we can deliver Go to row functionality to at least match what's already in Swing.

Reply:
Ahhh okay I've got it. With a table like:
```
t = empty_table(10_000).update(["x=i", "y=i*0.1", "z=i*1.0", "r=Math.random()*10_000"]).sort("r")
```
Then trying to seek to value `95` on column `Z` it ends up seeking to the row `9,499.0000` instead. Booo. Okay I'll limit it so you need to provide the value as the expected type I guess?